File: Python/ProjectEuler0006.py

# projecteuler.net/problem=6

# Answer: 25164150

# github.com/firatksee


def sumSqrDiff(start, stop):
    sumOfSqr = sum(map(lambda x: x**2, range(start, stop)))
    sqrOfSum = sum(range(start, stop)) ** 2

    return abs(sumOfSqr - sqrOfSum)


if __name__ == "__main__":
    print(sumSqrDiff(1, 101))

File: JavaScript/ProjectEuler0006.js
// projecteuler.net/problem=6

// Answer: 25164150

// github.com/firatksee

function sumSqrDiff(start, stop) {
    let sumOfSqr = 0;
    let sum = 0;

    for (let i = start; i < stop; i++) {
        sumOfSqr += i * i;
        sum += i;
    }

    const sqrOfSum = sum * sum;

    return Math.abs(sumOfSqr - sqrOfSum);
}

console.log(sumSqrDiff(1, 101));
